onnx_caffe2/backend.py

```diff
@@ -288,6 +288,9 @@ def _create_pad(cls, n, opset_version):
                 # first two dim is for batch and channel
                 set(pads[:2] + pads[4:6]) == {0}):
             raise ValueError('Caffe2 only supports padding 2D Tensor, whereas padding is ' + str(pads))
+        # Guard the invalid (negative) pads attribute.
+        if min(set(pads)) < 0:
+            raise ValueError('ONNX does not support negative pads in Pad, but get {}.'.format(pads))
```

Comment:
Please create an issue in the ONNX repo for kicking off the discussion of adding per-op validation.

Reply:
https://github.com/onnx/onnx/issues/362